examples/large_models/codegen25_ipex/ipex_woq/codegen_handler.py

```diff
@@ -0,0 +1,389 @@
+                torch.zeros([1, n_heads, 1, head_dim]).contiguous(),
+                beam_idx_tmp,
+            )
+            for i in range(n_layers)
+        ]
+
+        logger.info(f"num_attention_heads: {n_heads}, num_hidden_layers: {n_layers}, hidden size: {hidden_size}, head_dim: {head_dim}")
+        
+        # lets implement the WOQ 
+        if self.ipex_weight_only_quantization:
+            weight_dtype = torch.quint4x2 if self.woq_dtype == "INT4" else torch.qint8
+
+            if self.lowp_mode == "INT8":
+                lowp_mode = ipex.quantization.WoqLowpMode.INT8
+            elif self.lowp_mode == "FP32":
+                lowp_mode = ipex.quantization.WoqLowpMode.NONE
+            elif self.lowp_mode == "FP16":
+                lowp_mode = ipex.quantization.WoqLowpMode.FP16
+            elif self.lowp_mode == "BF16":
+                lowp_mode = ipex.quantization.WoqLowpMode.BF16
+            else:
+                lowp_mode = ipex.quantization.WoqLowpMode.BF16
+
+            act_quant_mode_dict = {
+                "PER_TENSOR": ipex.quantization.WoqActQuantMode.PER_TENSOR,
+                "PER_IC_BLOCK": ipex.quantization.WoqActQuantMode.PER_IC_BLOCK,
+                "PER_BATCH": ipex.quantization.WoqActQuantMode.PER_BATCH,
+                "PER_BATCH_IC_BLOCK": ipex.quantization.WoqActQuantMode.PER_BATCH_IC_BLOCK,
+            }
+
+            qconfig = ipex.quantization.get_weight_only_quant_qconfig_mapping(
+                weight_dtype=weight_dtype,
+                lowp_mode=lowp_mode,
+                act_quant_mode=act_quant_mode_dict[self.act_quant_mode],
+                group_size=self.group_size,
+            )
+            
+            # low precision checkpoint can be loaded, but we're considering there isn't any 
+            low_precision_checkpoint = None
+            self.user_model = ipex.llm.optimize(
+                self.user_model.eval(),
+                dtype=self.amp_dtype,
+                quantization_config=qconfig,
+                inplace=True,
+                low_precision_checkpoint=low_precision_checkpoint,
+                deployment_mode=False,
+            )
+            logger.info("The model conversion completed, now tracing the quantized model")
+
+            example_inputs = self.get_example_inputs()
+
+            with torch.no_grad(), torch.cpu.amp.autocast(
+                enabled=self.amp_enabled,
+                dtype=self.amp_dtype
+            ):
+                self_jit = torch.jit.trace(self.user_model.eval(), example_inputs, strict=False, check_trace=False)
+                self_jit = torch.jit.freeze(self_jit.eval())
+
+                self_jit.save(self.quantized_model_path)
+
+            logger.info("The IPEX Weight only quantization has been completed successfully")
+            
+        # set PAD token
+        if self.tokenizer.pad_token is None:
+            self.tokenizer.pad_token=self.tokenizer.eos_token
+        
+
+        if self.benchmark:
```

Comment:
In TS, the initialize function is used to load model. Here, the benchmark is trying to run inference based on the sample input. TS supports customized metrics in backend to measure each stage latency during inference. So this section is not needed.

Reply:
Removed benchmark and benchmark-related args.